packages/shared/src/porter.ts

```diff
@@ -120,10 +156,39 @@ export type TacoDecryptResult = {
 };
 
 export class PorterClient {
-  readonly porterUrl: URL;
+  readonly porterUrls: URL[];
 
-  constructor(porterUri: string) {
-    this.porterUrl = new URL(porterUri);
+  constructor(porterUris: string | string[]) {
+    if (porterUris instanceof Array) {
+      this.porterUrls = porterUris.map((uri) => new URL(uri));
+    } else {
+      this.porterUrls = [new URL(porterUris)];
+    }
+  }
+
+  protected async tryAndCall<T, D>(
+    config: AxiosRequestConfig<D>,
+  ): Promise<AxiosResponse<T>> {
+    let resp!: AxiosResponse<T>;
+    let lastError = undefined;
+    for (const porterUrl of this.porterUrls) {
+      const localConfig = { ...config, baseURL: porterUrl.toString() };
+      try {
+        resp = await axios.request(localConfig);
+      } catch (e) {
+        lastError = e;
+        continue;
+      }
+      if (resp.status === HttpStatusCode.Ok) {
+        return resp;
+      }
+    }
+    if (lastError !== undefined) {
+      throw lastError;
+    }
+    throw new Error(
+      'Porter returns bad response: ${resp.status} - ${resp.data}',
```

Comment:
Do we ever test for throwing this error?

```suggestion
      `Porter returns bad response: ${resp.status} - ${resp.data}`,
```

Reply:
Yes, there is test in porter.test `returns error in case all porters fail`

Reply:
Ok, so we didn't catch this issue (template string misuse) because `returns error in case all porters fail` doesn't test for the error message in the `Error` we've thrown

Reply:
✔️